specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2020-03-01/machineLearningServices.json

```
@@ -300,6 +300,88 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/listresourcepreparationerrors": {
```

what is the need for this API? cant the error be listed as part of the response to /prepareNotebook action?
It seems like there are actions being added for an entity which could be "notebook". Do you think there will be value in adding an "notebook" resource type under workspaces?
If you need this GET, then it can be called notebookPreparationErrors. SO you would remove the list from the name. But I doubt you need this separate type.Since prepare notebook is an action on workspaces, you can have a property on workspaces which indicate the notebookPrepareErrors. User can do a GET on /workspace to view it.

---

Also this URI is causing spell CI check failures. I suggest that we could make this in camel cases?
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/listResourcePreparationErrors": {
```

---

The reason we need this API is, we will try to prepare a notebook for user when a workspace is created, but we don't want to fail the workspace creation. So we need another prepareNotebook action to let user retry themselves. The reason we didn't make notebook a sub resource is the notebook name can not be defined by user, we are generating it on behalf of the user.

Also, this API will not only be used for displaying notebook preparation errors. We are planning to use it for other preparation errors to come. Notebook errors is just a starting point. That's why we didn't name the api "notebookPreparationErrors"

---

Actually I was rethinking this, it makes more sense to have a property for notebookPrepareErrors in GetWorkspace because it will save a call to separately retrieve the errors. Will make that change